pkgs/by-name/mo/monophony/package.nix

```
@@ -0,0 +1,80 @@
+{ lib
+, fetchFromGitLab
+, python3Packages
+, wrapGAppsHook4
+, gst_all_1
+, gobject-introspection
+, gtk4
+, yt-dlp
+, libadwaita
+, libsoup_3
+, glib-networking
+,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "monophony";
+  version = "2.3.0";
+  format = "other";
+
+  sourceRoot = "source/source";
+  src = fetchFromGitLab {
+    owner = "zehkira";
+    repo = "monophony";
+    rev = "v${version}";
+    hash = "sha256-n9tioIXMjUs9AmNuVLrOHJqeGhotkFlv+bTeHlsQPaM=";
+  };
+
+  pythonPath = with python3Packages; [
+    brotli
+    gst-python
+    mutagen
+    ordered-set
+    pycryptodomex
+    pygobject3
+    requests
+    websockets
+    ytmusicapi
+    mpris_server
```

```suggestion
    ordered-set
    ytmusicapi
    mpris_server
```
can trim this list down to the toplevel modules specified in [python3-modules.json](https://gitlab.com/zehkira/monophony/-/blob/3e3c685aa38b086287b1b672de05484c35fe6506/source/data/python3-modules.json) sans yt-dlp and nuitka which are specified elsewhere.

---

Sorry, how?

---

Like, how can we just remove so many deps

---

gst-python isn't needed as that comes via the gi interface (pygobject3). the rest are dependencies for imported modules, and I think as long as they are listed in the propagatedBuildInputs they will be included by this module.

`ordered-set` is just needed by the nitka build, so that can be moved to `natvieBuildInputs` and `pygobject3` should be added back, even though it comes in from mpris_server, it is a direct import of monophony so should be included.

---

Okay I'll do it tomorrow :)

---

Scrap that, I'll do it tomorrow xD

---

Had exams today

---

I think this is done